core/manifest/ClassManifest.php

```diff
@@ -333,6 +360,123 @@ public function handleDir($basename, $pathname, $depth) {
 		}
 	}
 
+	/**
+	 * Find a the full namespaced declaration of a class (or interface) from a list of candidate imports
+	 *
+	 * This is typically used to determine the full class name in classes that have imported namesapced symbols (having
+	 * used the `use` keyword)
+	 *
+	 * NB: remember the '\\' is an escaped backslash and is interpreted as a single \
+	 *
+	 * @param string $class The class (or interface) name to find in the candidate imports
+	 * @param string $namespace The namespace that was declared for the classes definition (if there was one)
+	 * @param array $imports The list of imported symbols (Classes or Interfaces) to test against
+	 *
+	 * @return string The fully namespaced class name
+	 */
+	protected function findClassOrInterfaceFromCandidateImports($class, $namespace = '', $imports = array()) {
+
+		//normalise the namespace
+		$namespace = rtrim($namespace, '\\');
+
+		//by default we'll use the $class as our candidate
+		$candidateClass = $class;
+
+		if ($class) {
+			//if the class starts with a \ then it is explicitly in the global namespace and we don't need to do
+			// anything else
+			if (substr($class, 0, 1) == '\\') {
+				$candidateClass = substr($class, 1);
+			}
+			else {
+				//if there's a namespace, starting assumption is the class is defined in that namespace
+				if ($namespace) {
+					$candidateClass = $namespace . '\\' . $class;
+				}
+
+				if (!empty($imports)) {
+
+					//normalised class name (PHP is case insensitive for symbols/namespaces
+					$lClass = strtolower($class);
+
+					//go through all the imports and see if the class exists within one of them
+					foreach ($imports as $alias => $import) {
+						//normalise import
+						$import = trim($import, '\\');
+
+						//if there is no string key, then there was no declared alias - we'll use the main declaration
+						if (is_int($alias)) {
```

Comment:
Maybe some more continue, break, or break out some of these inner loops into functions for readability?